README.md

```diff
@@ -1,62 +1,164 @@
     async def some_async_fn():
        ...
-       
-       
+```
+
+`CoolSyncClass` can be called synchronously:
+
+```python
 aaa = CoolSyncClass().some_async_fn()
 ```
 
+It can also be called asynchronously by passing `sync=False` and using `await`:
+
+```python
+aaa = await CoolSyncClass().some_async_fn(sync=False)
 ```
+
+You can also create a class that can be executed in both synchronous and asynchronous contexts by not setting the `asynchronous` attribute and passing it as an argument when creating an instance:
+
+```python
 class CoolDualClass(ASyncGenericBase):
     def __init__(self, asynchronous):
         self.asynchronous=asynchronous
     
     async def some_async_fn():
        ...
-       
+```
 
-async_instance = CoolDualClass(asynchronous=True)
-sync_instance = CoolDualClass(asynchronous=False)
+You can create an instance of `CoolDualClass` with `asynchronous=True` to call it asynchronously:
 
+```python
+async_instance = CoolDualClass(asynchronous=True)
 aaa = await async_instance.some_async_fn()
 aaa = async_instance.some_async_fn(sync=True)
+```
 
+You can also create an instance with `asynchronous=False` to call it synchronously:
+
+```python
+sync_instance = CoolDualClass(asynchronous=False)
 aaa = sync_instance.some_async_fn()
 aaa = sync_instance.some_async_fn(sync=False)
 ```
 
+### Settings
+
+The `ez-a-sync` library provides several settings that can be used to customize the behavior of the decorators and classes.
+
+To apply settings to the decorators or base classes, simply pass them as keyword arguments when calling the decorator or creating an instance.
+
+For example, to apply `cache_type='memory'` to a function decorated with `@a_sync('async')`, you would do the following:
+
+```python
+@a_sync('async', cache_type='memory')
+def some_function():
+    ...
+```
+
+#### async modifiers
+The `@a_sync('async')` decorator has the following settings:
+- `cache_type`: This can be set to `None` or `'memory'`. `'memory'` is a LRU cache which can be modified with the `cache_typed`, `ram_cache_maxsize`, and `ram_cache_ttl` modifiers.
+- `cache_typed`: Set to `True` if you want types considered treated for cache keys. i.e. with `cache_typed=True`, `Decimal(0)` and `0` will be considered separate keys.
+- `ram_cache_maxsize`: The maxsize for your LRU cache. Set to `None` if the cache is unbounded. If you set this value without specifying a cache type, `'memory'` will automatically be applied.
+- `ram_cache_ttl`: The TTL for items in your LRU cache. Set to `None`. If you set this value without specifying a cache type, `'memory'` will automatically be applied.
+- `runs_per_minute`: Setting this value enables a rate limiter for the decorated function.
+- `semaphore`: Drop in a Semaphore for your async defined functions.
+
+#### sync modifiers
+The `@a_sync('sync')` decorator has the following setting:
+- `executor`: The executor for the synchronous function. Set to the library's default of `config.default_sync_executor`.
+
```

Comment:
SHoudl add a secion on settings, which are different. will be easiest to just chat this one thru

Reply:
setings tldr are just user-specified defaults that will be applied as modifiers when no modifier value is passed